From 525757de852823f6a8df90e17bfbafd15f70ce3d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Lothar=20Wa=C3=9Fmann?= <LW@KARO-electronics.de>
Date: Thu, 16 Jan 2020 12:22:43 +0100
Subject: [PATCH 21/35] net: phy: micrel: fix the max value that is used to cap
 the skew register values

The maximum value that may be written to a register depends on the
register being accessed and is should be calculated from the bitfield
size rather than fixed at 0xf.
---
 drivers/net/phy/micrel_ksz90x1.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/drivers/net/phy/micrel_ksz90x1.c b/drivers/net/phy/micrel_ksz90x1.c
index df58562..7a8ea3c 100644
--- a/drivers/net/phy/micrel_ksz90x1.c
+++ b/drivers/net/phy/micrel_ksz90x1.c
@@ -127,13 +127,13 @@ static int ksz90x1_of_config_group(struct phy_device *phydev,
 			regval |= ofcfg->grp[i].dflt << offset;
 		} else {
 			changed = 1;	/* Value was changed in OF */
+			max = (1 << ofcfg->grp[i].size) - 1;
+
 			/* Calculate the register value and fix corner cases */
-			if (val > ps_to_regval * 0xf) {
-				max = (1 << ofcfg->grp[i].size) - 1;
+			if (val > ps_to_regval * max)
 				regval |= max << offset;
-			} else {
+			else
 				regval |= (val / ps_to_regval) << offset;
-			}
 		}
 	}
 
-- 
2.7.4

